pvlib/modelchain.py

```diff
@@ -276,14 +280,29 @@ class ModelChainResult:
     tracking: Optional[pd.DataFrame] = field(default=None)
     total_irrad: Optional[PerArray[pd.DataFrame]] = field(default=None)
     aoi: Optional[PerArray[pd.Series]] = field(default=None)
-    aoi_modifier: Optional[PerArray[pd.Series]] = field(default=None)
-    spectral_modifier: Optional[PerArray[pd.Series]] = field(default=None)
+    aoi_modifier: Optional[PerArray[Union[pd.Series, float]]] = \
+        field(default=None)
+    spectral_modifier: Optional[PerArray[Union[pd.Series, float]]] = \
+        field(default=None)
     cell_temperature: Optional[PerArray[pd.Series]] = field(default=None)
     effective_irradiance: Optional[PerArray[pd.Series]] = field(default=None)
     dc: Optional[PerArray[Union[pd.Series, pd.DataFrame]]] = \
         field(default=None)
     diode_params: Optional[PerArray[pd.DataFrame]] = field(default=None)
 
+    def _result_type(self, value):
+        # Allow None to pass through without being wrapped in a tuple
+        if (self._singleton_tuples
+                and not isinstance(value, tuple)
+                and value is not None):
+            return (value,)
+        return value
+
+    def __setattr__(self, key, value):
+        if key in ModelChainResult._per_array_fields:
```

Comment:
could we use [`typing.get_type_hints`](https://docs.python.org/3/library/typing.html#typing.get_type_hints) and delete the private set?

Reply:
That would be nice, but I can't seem to figure out how to compare the concrete types returned from `get_type_hints` with the generic `Optional[PerArray[_T]]` type.

Reply:
After further research, this seems like a misuse of the typing module at this point in time, so I'm good with your solution.

Reply:
I'm a little bummed... I was excited to actually put the types to use.